intel_extension_for_transformers/llm/operator/csrc/include/dropout.hpp

```diff
@@ -0,0 +1,159 @@
+//  Copyright (c) 2023 Intel Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+#include "../dispatcher/include/jblas_weightonly_dispatcher.hpp"
+#include "jblas/jit_blas_utils.h"
+#include <ATen/core/TensorBody.h>
+#include <cstddef>
+#include <cstdint>
+#include <ctime>
+#include <immintrin.h>
+#include <iostream>
+#include <omp.h>
+#include <torch/script.h>
+#include <torch/torch.h>
+#include <torch/types.h>
+#include <vector>
+
+class RandBuffer {
+ public:
+  RandBuffer() {
+    std::srand((int)std::time(0));
+    auto thread_num = omp_get_max_threads();
+    load_buffer = jblas::utils::amalloc<uint32_t>(thread_num * 16);
+    iws = jblas::utils::amalloc<int>(thread_num * 16);
+    for (int i = 0; i < thread_num; i++) initMWC(rand(), i);
+  }
+
+  ~RandBuffer() {
+    if (iws != NULL) jblas::utils::afree(iws);
+    if (load_buffer != NULL) jblas::utils::afree(load_buffer);
+  }
+
+#pragma GCC push_options
+#pragma GCC target("avx512f", "avx512bw", "avx512vl", "avx512vbmi", "avx512dq")
```

Comment:
maybe invoke assembly directly will be better?

Reply:
You are right, this approach avoids frequent push/pop compilation options and has better compiler version compatibility. Let's modify it in the next PR, we need to merge this pr asap to avoid blocking xinyu's task.